lib/SIL/Verifier/LinearLifetimeChecker.cpp

```diff
@@ -485,35 +482,30 @@ void State::checkDataflowEndState(DeadEndBlocks &deBlocks) {
     // wants us to tell it where to insert compensating destroys.
   }
 
-  // Make sure that we do not have any lifetime ending uses left to visit that
-  // are not transitively unreachable blocks.... so return early.
-  if (blocksWithNonConsumingUses.empty()) {
-    return;
-  }
-
   // If we do have remaining blocks, then these non lifetime ending uses must be
   // outside of our "alive" blocks implying a use-after free.
-  for (auto &pair : blocksWithNonConsumingUses) {
-    if (deBlocks.isDeadEnd(pair.first)) {
+  for (auto pair : blocksWithNonConsumingUses.getRange()) {
```

Comment:
NOTE: This range skips "blotted" (Key, [Value]).

Reply:
That is, we represent erased values differently, so blocksWithNonConsumingUses will /never/ actually be empty except in trivial cases. Instead we will have arrays filled with blotted elements that we skip. So just having the loop here is correct.